osu.Server.Spectator/Hubs/ScoreUploader.cs

```diff
@@ -103,6 +107,17 @@ private async Task readLoop()
                         if (!dbScore.passed)
                             continue;
 
+                        if (dbScore.build_id != null)
```

Comment:
I would have just joined on the existing `db.GetScoreFromTokenAsync` query, but I guess this will work.